platzky/engine.py

```
@@ -82,62 +122,55 @@ def add_health_check(self, name: str, check_function: Callable[[], None]) -> Non
+        health_check_timeout = 10  # seconds
+
+        def run_health_check(
+            executor: ThreadPoolExecutor,
+            check_func: Callable[[], None],
+            timeout: int,
+        ) -> str:
+            """Run a health check with timeout, returning status string."""
+            future = executor.submit(check_func)
+            try:
+                future.result(timeout=timeout)
+                return "ok"
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)